examples/bioasq.py

```diff
@@ -0,0 +1,579 @@
+exact and ideal answers.
+"""
+
+_BIOASQ_4B_DESCRIPTION = """\
+The data are intended to be used as training and development data for BioASQ 4,
+which will take place during 2016. There is one file containing the data:
+ - BioASQ-trainingDataset4b.json
+
+The file contains the data of the first three editions of the challenge: 1307
+questions with their relevant documents, snippets, concepts and rdf triples,
+exact and ideal answers from the first two editions and 497 questions with
+similar annotations from the third editions of the challenge.
+"""
+
+_BIOASQ_3B_DESCRIPTION = """No README provided."""
+
+_BIOASQ_2B_DESCRIPTION = """No README provided."""
+
+_DESCRIPTION = {
+    "bioasq10b": _BIOASQ_10B_DESCRIPTION,
+    "bioasq9b": _BIOASQ_9B_DESCRIPTION,
+    "bioasq8b": _BIOASQ_8B_DESCRIPTION,
+    "bioasq7b": _BIOASQ_7B_DESCRIPTION,
+    "bioasq6b": _BIOASQ_6B_DESCRIPTION,
+    "bioasq5b": _BIOASQ_5B_DESCRIPTION,
+    "bioasq4b": _BIOASQ_4B_DESCRIPTION,
+    "bioasq3b": _BIOASQ_3B_DESCRIPTION,
+    "bioasq2b": _BIOASQ_2B_DESCRIPTION,
+}
+
+_HOMEPAGE = "http://participants-area.bioasq.org/datasets/"
+
+# Data access reqires registering with BioASQ.
+# See http://participants-area.bioasq.org/accounts/register/
+_LICENSE = ""
+
+_URLs = {
+    "bioasq10b": ["BioASQ-training10b.zip", None],
+    "bioasq9b": ["BioASQ-training9b.zip", "Task9BGoldenEnriched.zip"],
+    "bioasq8b": ["BioASQ-training8b.zip", "Task8BGoldenEnriched.zip"],
+    "bioasq7b": ["BioASQ-training7b.zip", "Task7BGoldenEnriched.zip"],
+    "bioasq6b": ["BioASQ-training6b.zip", "Task6BGoldenEnriched.zip"],
+    "bioasq5b": ["BioASQ-training5b.zip", "Task5BGoldenEnriched.zip"],
+    "bioasq4b": ["BioASQ-training4b.zip", "Task4BGoldenEnriched.zip"],
+    "bioasq3b": ["BioASQ-trainingDataset3b.zip", "Task3BGoldenEnriched.zip"],
+    "bioasq2b": ["BioASQ-trainingDataset2b.zip", "Task2BGoldenEnriched.zip"],
+}
+
+_SUPPORTED_TASKS = ["qa"]
+_SOURCE_VERSION = datasets.Version("1.0.0")
+_BIOBIO_VERSION = datasets.Version("1.0.0")
+
+
+class BioasqDataset(datasets.GeneratorBasedBuilder):
+    """
+    BioASQ Task B On Biomedical Semantic QA.
+    Creates configs for BioASQ2 through BioASQ10.
+    """
+
+    VERSION = datasets.Version("1.0.0")
+    DEFAULT_CONFIG_NAME = "bioasq9b_source"
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(
+            name=f"bioasq{vers}b_source",
+            version=_SOURCE_VERSION,
+            description=f"bioasq{vers} Task B source schema",
+        )
+        for vers in range(2, 11)
+    ]
+    BUILDER_CONFIGS += [
+        datasets.BuilderConfig(
+            name=f"bioasq{vers}b_bigbio",
+            version=_BIOBIO_VERSION,
+            description=f"bioasq{vers} Task B in simplified BigBio schema",
+        )
+        # BioASQ2 through BioASQ10
+        for vers in range(2, 11)
+    ]
+
+    def _info(self):
+
+        task_version, schema = self.config.name.split("_")
```

Comment:
I took a pass at the custom BuilderConfig
```
@dataclass
class BigBioConfig(datasets.BuilderConfig):
    """BuilderConfig for BigBio."""
    schema: str = None
    tasks: list = field(default_factory=list)
```
However, there are a few issues that undermines the utility here. Maybe we would want something like this when loading datasets
```
data = load_dataset(fpath, name="bioasq9b", schema="bigbio")
```
However, the `name` field must be unique across dataloader, so we're forced into always using something like `bioasq9b_source` or `bioasq9b_biobio_qa` regardless. That's fine, we could still use the `schema` property to create a more elegant way to detect schema without regex or string munging. 

However the `tasks` list seems less useful to me vs a `_SUPPORTED_TASKS` var in the script. We want that for unit tests and (eventually) the prompting stuff. You also can't seem to access the dataset config outside of the internal `load_dataset` building loop so that creates other problems. 

I think we should just run with `bioasq7b_bigbio-qa` string of the format `DATASET_SCHEMA' where SCHEMA is at the main type of schema (e.g., `bc5cdr_bigbio-kb`) since a single BuildConfig/schema can map to multiple tasks